test/runtime/index.ts

```diff
@@ -203,21 +203,20 @@ describe('runtime', () => {
 					}
 
 					if (config.test) {
-						return Promise.resolve(config.test({
+						config.test({
 							assert,
 							component,
 							mod,
 							target,
 							window,
 							raf,
 							compileOptions
-						})).then(() => {
-							component.$destroy();
-
-							if (unhandled_rejection) {
-								throw unhandled_rejection;
-							}
 						});
+						component.$destroy();
+
+						if (unhandled_rejection) {
+							throw unhandled_rejection;
+						}
```

Comment:
Is it intentional that we remove the `return Promise.resolve` here? Isn't `config.test()` async?

Reply:
Super catch!
Sorry, I revert it.😰
https://github.com/sveltejs/svelte/pull/7076/commits/6b95912dd70743e83e2a64ceb70c81ad8deb5b45